privacyidea/api/lib/postpolicy.py

```diff
@@ -318,6 +318,54 @@ def no_detail_on_success(request, response):
     return response
 
 
+def preferred_client_mode(request, response):
+    """
+    This policy function is used to add the preferred client mode.
+    The admin can set the list of client modes in the policy in the
+    same order as  he preferred them. The faction will pick the first
+    client mode from the list, that is also in the multichallege and
+    set it as preferred client mode
+
+    :param request:
+    :param response:
+    :return:
+    """
+    content = response.json
+
+    # get the preferred client mode from a policy definition
+    detail_pol = Match.action_only(g, scope=SCOPE.AUTH, action=ACTION.PREFERREDCLIENTMODE)\
+        .action_values(unique=True)
+
+    if detail_pol:
+        preferred_client_mode_list = detail_pol.get("test")[0].split(", ")
```

Comment:
Lists in policies are whitespace separated.
So an admin would configure the policy like:

      "webauthn poll"

Hm, what is this with `get("test")`. THis does not make any sense.
Please see the policy `challenge_response` (https://github.com/privacyidea/privacyidea/blob/9bee74fb74a492ee97ca4ee9ec147312f8800108/privacyidea/lib/policydecorators.py#L122) to understand values in one policy.

Hm, it is still not clear, how we should handle multiple policies in this case:

pol1:  webauthn poll
pol2:   pol

Reply:
~~~~python
Match.user().action_values(allow_white_space_in_action=True)
~~~~

`allow_white_space_in_action` would make a string like "poll interactive" *one* action value.
Otherwise it would be split into two action values "poll" and "interactive".

Reply:
[34ee918](https://github.com/privacyidea/privacyidea/pull/3361/commits/34ee918173549cdef4dfeb1e000951457fff1cab)